.circleci/config.yml

```diff
@@ -0,0 +1,50 @@
+# We are supposed to use architect-orb to build&push container images.
+# However, architect-orb has strict rules and tests. We don't have full control over this repository.
+# That is why we introduced this custom configuration in this repository.
+# Keep this file as it is while merging changes from upstream repository.
+
+version: 2.1
+
+jobs:
+  build:
+    machine:
+      image: ubuntu-2204:2023.02.1
+    resource_class: large
+    steps:
+    - checkout
+    - run:
+        name : Build container
+        command: |
+          # setup vars
+          export REGISTRY=gsoci.azurecr.io/giantswarm
+          export IMAGE_NAME=cluster-api-azure-controller
+          export ARCH=amd64
+          export TAG="${CIRCLE_TAG:-"${CIRCLE_SHA1}"}"
+          
+          # Build container image with upstream make docker-build
+          make docker-build
+    - run:
+        name: Push to gsoci.azurecr.io
```

Comment:
This was docker.io and quay.io in the previous release but we switched to azure registry. I changed it.